encord_agents/gcp/wrappers.py

```diff
@@ -49,10 +51,32 @@ def editor_agent(
 
     def context_wrapper_inner(func: AgentFunction) -> Callable[[Request], Response]:
         dependant = get_dependant(func=func)
+        cors_regex = re.compile(ENCORD_DOMAIN_REGEX)
 
         @wraps(func)
         def wrapper(request: Request) -> Response:
-            frame_data = FrameData.model_validate_json(orjson.dumps(request.form.to_dict()))
+            if request.method == "OPTIONS":
+                response = make_response("")
+                response.headers["Vary"] = "Origin"
+
+                if not cors_regex.fullmatch(request.origin):
+                    response.status_code = 403
+                    return response
+
+                headers = {
+                    "Access-Control-Allow-Origin": request.origin,
+                    "Access-Control-Allow-Methods": "POST",
+                    "Access-Control-Allow-Headers": "Content-Type",
+                    "Access-Control-Max-Age": "3600",
+                }
+                response.headers.update(headers)
+                response.status_code = 204
+                return response
+
+            if request.is_json:
+                frame_data = FrameData.model_validate(request.get_json())
+            else:
```

Comment:
Is this necessary or just defensive?
Is it worth having a try/ except around these conditions given that it could throw?

Reply:
Added the TODO